frontend/packages/console-app/src/components/detect-perspective/PerspectiveDetector.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import { useExtensions, isPerspective } from '@console/plugin-sdk';
+
+type PerspectiveDetectorProps = {
+  setActivePerspective: (string) => void;
+};
+
+const PerspectiveDetector: React.FC<PerspectiveDetectorProps> = ({ setActivePerspective }) => {
+  let detectedPerspective: string;
+  let detectionComplete: boolean;
+  const perspectiveExtensions = useExtensions(isPerspective);
+  const defaultPerspective = perspectiveExtensions.find((p) => p.properties.default);
+  const detectorPerspectives = perspectiveExtensions.filter(
+    (perspective) => perspective.properties.usePerspectiveDetection,
+  );
+
+  detectorPerspectives.some((perspective, index) => {
+    const [enablePerspective, loading] = perspective.properties.usePerspectiveDetection();
+
+    if (!loading && enablePerspective) {
+      detectedPerspective = perspective.properties.id;
+      detectionComplete = !loading;
+      return true;
+    }
+
+    if (detectorPerspectives.length === index + 1) detectionComplete = !loading;
```

Comment:
This doesn't seem right.
You need to check that _every_ detector has completed loading and each has resulted in a value of `false` for the detection.

Reply:
Changed the logic to take this into consideration.